eng/mgmt/automation/typespec_utils.py

```diff
@@ -0,0 +1,67 @@
+import os
+import re
+import sys
+import yaml
+import logging
+
+
+def validate_tspconfig(tsp_dir: str) -> bool:
+    valid = True
+
+    if not tsp_dir.endswith('tspconfig.yaml'):
+        tsp_dir = os.path.join(tsp_dir, 'tspconfig.yaml')
+    with open(tsp_dir, 'r') as tspconfig:
+        yaml_json = yaml.safe_load(tspconfig)
+
+    service_dir_pattern = r'sdk/\w+'
+    package_dir_pattern = r'azure(-\w+)+'
+    namespace_pattern = r'com\.azure(\.\w+)+'
+
+    # SDK automation would make sure these properties exists
+    service_dir: str = yaml_json['parameters']['service-dir']['default']
```

Comment:
Will this throw if `service-dir` does not exist?

Reply:
I will add check for `namespace`, which may not be guaranteed to exist by toolchain.